src/Agent.Plugins/Artifact/FileContainerProvider.cs

```diff
@@ -427,7 +427,14 @@ private void ExtractTarsIfPresent(AgentTaskPluginExecutionContext context, IEnum
 
                     ExtractTar(fileArtifactPath, extractedFilesDir);
 
-                    File.Delete(fileArtifactPath);
+                    try
+                    {
+                        File.Delete(fileArtifactPath);
+                    }
+                    catch
+                    {
+                        context.Warning($"Unable to delete artifact files");
```

Comment:
I think that we should catch specific error (and also output it with reason, e.g: File is used by another process)